src/hooks/useIsSnapInstalled/useIsSnapInstalled.tsx

```diff
@@ -39,16 +39,17 @@ export const useIsSnapInstalled = (): null | boolean => {
   const [isSnapInstalled, setIsSnapInstalled] = useState<null | boolean>(null)
 
   const {
-    state: { wallet },
+    state: { wallet, isConnected, isDemoWallet },
   } = useWallet()
 
   const checkSnapInstallation = useCallback(async () => {
+    if (!isConnected || isDemoWallet) return
```

Comment:
While at it - additional safety not to spam JSON-RPC requests, and not even begin the `checkIsMetaMask` checks until we have a connected wallet